From 5d202d481c3ba44a29df66789bb58e183513b818 Mon Sep 17 00:00:00 2001
From: "Munoz, Obed N" <obed.n.munoz@intel.com>
Date: Wed, 8 Jul 2015 13:53:57 -0500
Subject: [PATCH] Fix date parsing when there's not milliseconds in the date

When OSProfiler is converting date string to datetime format
"%Y-%m-%dT%H:%M:%S.%f", it will raise a ValueError exception if it
receives dates with the following format: "%Y-%m-%dT%H:%M:%S"

The "%Y-%m-%dT%H:%M:%S" date format applies when a notification
occurs in the exact 0 millisecond of a certain second.

Adding a try-except section for catching the exceptions when the date
format is "%Y-%m-%dT%H:%M:%S". It will try first to match with
"%Y-%m-%dT%H:%M:%S.%f" format and if it fails, it will go with the
non-milliseconds date format matching.

Closes-Bug: #1472735

Change-Id: I3fd0c3a3b26797f215d2ae7a8acc29c8619c0b68
Signed-off-by: Munoz, Obed N <obed.n.munoz@intel.com>
---
 osprofiler/parsers/ceilometer.py | 8 ++++++--
 1 file changed, 6 insertions(+), 2 deletions(-)

diff --git a/osprofiler/parsers/ceilometer.py b/osprofiler/parsers/ceilometer.py
index a49bede..ec6fe07 100644
--- a/osprofiler/parsers/ceilometer.py
+++ b/osprofiler/parsers/ceilometer.py
@@ -74,8 +74,12 @@ def parse_notifications(notifications):
             if k not in skip_keys:
                 result[key]["info"][k] = meta[k]
 
-        timestamp = datetime.datetime.strptime(n["timestamp"],
-                                               "%Y-%m-%dT%H:%M:%S.%f")
+        try:
+            timestamp = datetime.datetime.strptime(n["timestamp"],
+                                                   "%Y-%m-%dT%H:%M:%S.%f")
+        except ValueError:
+            timestamp = datetime.datetime.strptime(n["timestamp"],
+                                                   "%Y-%m-%dT%H:%M:%S")
 
         if meta["name"].endswith("stop"):
             result[key]["info"]["finished"] = timestamp
-- 
2.4.5

